tests/test_api.py

```diff
@@ -37,6 +38,16 @@ def test_property_result_is_path(func_path: str) -> None:
     assert isinstance(result, Path)
 
 
+def test_function_interface_is_in_sync(func: str) -> None:
+    function_dir = getattr(platformdirs, func)
+    function_path = getattr(platformdirs, func.replace("_dir", "_path"))
+    assert inspect.isfunction(function_dir)
+    assert inspect.isfunction(function_path)
+    function_dir_signature = inspect.Signature.from_callable(function_dir)
+    function_path_signature = inspect.Signature.from_callable(function_path)
+    assert function_dir_signature.parameters == function_path_signature.parameters
```

Comment:
We should also assert on the type annotations here 😊

Reply:
Do you mean that we need to assert that the type annotation of the function signatures are equal between both functions, or do you want to assert against an expected set of annotations? The former is already happening:
```py
>>> import inspect
>>> import pprint
>>> import platformdirs
>>> pprint.pprint(functools.inspect(platformdirs.user_data_path))
>>> pprint.pprint(inspect.Signature.from_callable(platformdirs.user_data_path).parameters)
mappingproxy(OrderedDict([('appname',
                           <Parameter "appname: Optional[str] = None">),
                          ('appauthor',
                           <Parameter "appauthor: Union[str, NoneType, ForwardRef('Literal[False]')] = None">),
                          ('version',
                           <Parameter "version: Optional[str] = None">),
                          ('roaming', <Parameter "roaming: bool = False">)]))
```
And [`Parameter.__eq__` compares the annotation](https://github.com/python/cpython/blob/3.9/Lib/inspect.py#L2616).